lighthouse-core/gather/gather-runner.js

```diff
@@ -115,6 +115,24 @@ class GatherRunner {
     return {};
   }
 
+  /**
+   * Reject if the gathering terminates due to the CDP target crashing, etc
+   * @param {Driver} driver
+   * @return {Promise<void>}
+   */
+  static async getGatherTerminatedPromise(driver) {
+    return new Promise((_, reject) => {
+      driver.on('Inspector.targetCrashed', _ => {
+        reject(new LHError(LHError.errors.TARGET_CRASHED));
+      });
+      // In case of crash, detached fires after targetCrashed, so we'll exit with the crash code
+      // Detachment happens (in non-crash cases) when the browser tab is closed or unexpected connection failure.
+      driver.on('Inspector.detached', _ => {
```

Comment:
Do we need to filter these to particular targets? i.e. I could see a frame detaching if it's removed or something, but I'm not sure

Reply:
so `Target.detachedFromTarget` is fired if an iframe/SW is removed.

but `Inspector.detached` is the special event that announces unexpected CDP detachments. it's what was introduced to improve the UX before multiclient CDP was available.

so nah we don't need to filter this one.